### PR TITLE
High-Frequency Machete

### PR DESCRIPTION
**Pull request by jknpj**
https://webmshare.com/omnZ0
https://webmshare.com/aDnP7
https://webmshare.com/jQ1QO
https://webmshare.com/59jaK

Adds High-Frequency Machete, an alternative all-purpuse weapon to the iconic energy sword, and its upgraded form, the High-Frequency Pincer Blade "Bloodlust". Both come with tons of unique sounds for enhanced murderbone experience. Sprites by @Blithering

machete2

The machete, unlike the energy sword it is always medium sized, can't be used as a shield and even with in high-frequency mode its force lower (25 against esword's 30) but  besides having higher sharpness (1.7) it also has other two fun gimmicks:

* When in high-frequency mode it'll instantly cut down plant segments(kudzu) as you move.
* When in normal mode it becomes a robust throwing weapon with 20 throwing force 8 speed and range.

bloodlust

Bloodlust was made with high risk high reward in mind. It's a hand-attached wieldable weapon with 2.5 seconds attack delay that packs whooping 34 force, 2 sharpness(really, really sharp.) and the ability to act as a shield, making it really strong but also really difficult to use and hide. You can use a screwdriver to detach it from your arm but picking it up attachs it again.

* It keeps its ability to instantly cut down plant segments when in high-frequency mode(wielded)
* Due being heavy, the bloodlust throwing force, speed range are always shit. Don't throw it.

:cl:
 * rscadd: Adds High-Frequency Machete to syndicate uplink, with a price of 8 TC each.
 * rscadd: Adds High-Frequency Pincer Blade "Bloodlust", it can be crafted using two high-frequency machetes.

### CODE REVIEW

**Comment by Aurx** [comment]
What role will this fill that an esword won't?
What fun things can be done with this but not an esword?

**Comment by jknpj** [comment]
@Aurx 

It isn't there to powercreep esword, it is an alternative since it fills a non-stealthy role but here we go:

The insta kudzu clean means you have a way to deal with traitor botanists/xenobotanists kudzuloose as well means you have a way to keep your own kudzu under control. 

The throwing gimmick is really good to finish off running targets  and thanks to our new grab feature if you're fast enough you can grab a machete throw at you, avoiding damage and turning the tables.

Bloodlust is slow but it's still a hell of a weapon. 2 sharpness is really robust and it allows you to cut limbs, damage organs and break bones with few hits. The thing that can't be easily hidden and you still can get disarmed and killed for being reckless but if you can manage to hit-and-run properly you can pull off some nice hostage situations or even cripple your foes without exactly going for the kill.

Overall i like how they turned out in the end. I still doubt it'll going to be used as much as the esword though.

**Comment by Aurx** [comment]
It's not that it's not distinct from the esword or that it's powercreepy.  
It's that I honestly don't see much design space for another melee weapon in the same price bracket.

File: code/game/objects/items/weapons/melee/energy.dm [hunk 1]
```diff
@@ -184,3 +184,98 @@
...
+
+/obj/item/weapon/melee/energy/hfmachete/throw_impact(atom/hit_atom)
+	if(isturf(hit_atom))
+		for(var/mob/M in hit_atom)
+			playsound(M, get_sfx("machete_throw_hit"),60, 0)
+	..()
+
+/obj/item/weapon/melee/energy/hfmachete/attack(target as mob, mob/living/user as mob)
+	if(active && istype(target, /obj/effect/plantsegment)) // really useful in a jungle..
+		var/obj/effect/plantsegment/P = istype(target, /obj/effect/plantsegment)
```
Comment (9600bauds) [inline code/game/objects/items/weapons/melee/energy.dm]: heck at least do `for plantsegment/P in range(1)` to make it more useful than a welder

**Comment by yclatious** [comment]
Being hit making you say "IM FUCKING INVINCIBLE" while wielding it when

File: code/game/objects/items/weapons/melee/energy.dm (hunk 1 above)
Comment (Shadowmech88) [reply code/game/objects/items/weapons/melee/energy.dm]: What about putting an event hook on it tied to a mob's movement that checks `for plantsegment/P in range(1)`, so you could cleave a path through kudzu just by walking through it while wielding the machete?

Comment (jknpj) [reply code/game/objects/items/weapons/melee/energy.dm]: Like this? https://webmshare.com/59jaK

Comment (Shadowmech88) [reply code/game/objects/items/weapons/melee/energy.dm]: Looks good to me.

**Comment by Blithering** [comment]
Maybe make the bloodlust mode have nodrop, since it's meant to be attached to a wrist-arm thing?

**Comment by jknpj** [comment]
Wouldn't it make the weapon too creatine-friendly?

**Comment by yclatious** [comment]
A nigh invincible hulked out man screaming in anger as he chops off people with a dual machete pincer type weapon sounds so good, I wouldn't even be mad if it happened to me.

**Comment by Shadowmech88** [comment]
Hulks can't be stunned or taken down anyway, what does it matter if a hulk's weapon has nodrop?

**Comment by Probe1** [comment]
I still think this is a silly addition but people want it.

**Pull request merged**